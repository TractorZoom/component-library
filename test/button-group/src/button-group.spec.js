import ButtonGroup from '../../../packages/button-group/src';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Button Group', () => {
    it('should render with a single button', () => {
        // given
        const mockData = {
            buttons: [{ props: { title: 'the title' } }],
        };

        // when
        const component = renderer.create(<ButtonGroup {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with a group of two buttons', () => {
        // given
        const mockData = {
            buttons: [{ props: { title: 'the title' } }, { props: { title: 'the title 2' } }],
        };

        // when
        const component = renderer.create(<ButtonGroup {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with a group of 10 buttons', () => {
        // given
        const mockData = {
            buttons: [
                { props: { title: 'the title' } },
                { props: { title: 'the title 2' } },
                { props: { title: 'the title 3' } },
                { props: { title: 'the title 4' } },
                { props: { title: 'the title 5' } },
                { props: { title: 'the title 6' } },
                { props: { title: 'the title 7' } },
                { props: { title: 'the title 8' } },
                { props: { title: 'the title 9' } },
                { props: { title: 'the title 10' } },
            ],
        };

        // when
        const component = renderer.create(<ButtonGroup {...mockData} />);
        let tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });
});
