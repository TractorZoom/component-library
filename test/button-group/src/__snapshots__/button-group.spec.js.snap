// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Group should render with a group of 10 buttons 1`] = `
<div
  className="makeStyles-buttonGroup-2"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 2
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 3
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 4
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 5
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 6
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 7
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 8
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 9
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 10
    </span>
  </button>
</div>
`;

exports[`Button Group should render with a group of two buttons 1`] = `
<div
  className="makeStyles-buttonGroup-2"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title 2
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button Group should render with a single button 1`] = `
<div
  className="makeStyles-buttonGroup-2"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      the title
    </span>
  </button>
</div>
`;
